scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -864,6 +864,8 @@ def _thread(handler, timeout, outdir, logfile, fifo_fn, pid_fn, results,
                         if cpu_time < timeout and not out_state:
                             timeout_time = time.time() + (timeout - cpu_time)
                             continue
+                except psutil.NoSuchProcess:
```

Comment:
Is it intentionally processed differently here? So on MacOS you will receive 'timeout' instead of 'failed' in the reason.

Reply:
It is intentionnal. I had several cases where the test succeeded but the process was closed before `_get_cpu_time` resulting in the exception being thrown out.
I suspect a ToCToU, but I'm not fluent enough in Python to figure exactly what is off.

This seemed like the best approximation of a fix.